contracts/src/v0.8/dev/OptimismValidator.sol

```diff
@@ -1,71 +1,185 @@
-  uint32 private constant CALL_GAS_LIMIT = 1_200_000;
+  address private immutable s_L2TokenBridgeAddr;
+  address private immutable s_L2ETHTokenAddr;
+
   int256 private constant ANSWER_SEQ_OFFLINE = 1;
+  uint32 private s_gasLimit;
+
+  address public immutable L1_CROSS_DOMAIN_MESSENGER_ADDRESS;
+  address public immutable L2_UPTIME_FEED_ADDR;
+  // L2 xDomain alias address of this contract
+  address public immutable L2_ALIAS = AddressAliasHelper.applyL1ToL2Alias(address(this));
+
+  AccessControllerInterface private s_configAC;
+
+  /**
+   * @notice emitted when a new gas access-control contract is set
+   * @param previous the address prior to the current setting
+   * @param current the address of the new access-control contract
+   */
+  event ConfigACSet(address indexed previous, address indexed current);
+
+  /**
+   * @notice emitted when a new ETH withdrawal from L2 was requested
+   * @param amount of funds to withdraw
+   */
+  event L2WithdrawalRequested(uint256 amount, address indexed refundAddr);
 
-  address public immutable CROSS_DOMAIN_MESSENGER;
-  address public immutable L2_CROSS_DOMAIN_FORWARDER;
-  address public immutable L2_FLAGS;
+  /**
+   * @notice emitted when gas cost to spend on L2 is updated
+   * @param gasLimit updated gas cost
+   */
+  event GasLimitUpdated(uint32 gasLimit);
 
   /**
-   * @param crossDomainMessengerAddr address the xDomain bridge messenger (Optimism bridge L1) contract address
-   * @param l2CrossDomainForwarderAddr the L2 Forwarder contract address
-   * @param l2FlagsAddr the L2 Flags contract address
+   * @param l1CrossDomainMessengerAddress address the L1CrossDomainMessenger contract address
+   * @param l2UptimeFeedAddr the address of the OptimismSequencerUptimeFeed contract address
+   * @param configACAddr address of the access controller for managing gas price on Optimism
    */
   constructor(
-    address crossDomainMessengerAddr,
-    address l2CrossDomainForwarderAddr,
-    address l2FlagsAddr
+    address l1CrossDomainMessengerAddress,
+    address l2UptimeFeedAddr,
+    address configACAddr,
+    address l2TokenBridgeAddr,
+    address l2ETHTokenAddr,
+    uint32 gasLimit
   ) {
-    require(crossDomainMessengerAddr != address(0), "Invalid xDomain Messenger address");
-    require(l2CrossDomainForwarderAddr != address(0), "Invalid L2 xDomain Forwarder address");
-    require(l2FlagsAddr != address(0), "Invalid L2 Flags address");
-    CROSS_DOMAIN_MESSENGER = crossDomainMessengerAddr;
-    L2_CROSS_DOMAIN_FORWARDER = l2CrossDomainForwarderAddr;
-    L2_FLAGS = l2FlagsAddr;
+    require(l1CrossDomainMessengerAddress != address(0), "Invalid xDomain Messenger address");
+    require(l2UptimeFeedAddr != address(0), "Invalid OptimismSequencerUptimeFeed contract address");
+    L1_CROSS_DOMAIN_MESSENGER_ADDRESS = l1CrossDomainMessengerAddress;
+    L2_UPTIME_FEED_ADDR = l2UptimeFeedAddr;
+    // Additional L2 payment configuration
+    _setConfigAC(configACAddr);
```

Comment:
Is this external access controller necessary? `setGasLimit` uses `onlyOwner` and not `onlyOwnerOrConfigAccess`